tests/cli/test_analyze.py

```diff
@@ -0,0 +1,313 @@
+    del ARGV["eth-address"]
+
+
+def test_mythx_client_default_no_username_arg():
+    ARGV["password"] = "bar"
+    assert_trial_client()
+    del ARGV["password"]
+
+
+def test_mythx_client_default_no_password_env(monkeypatch):
+    monkeypatch.setenv("MYTHX_ETH_ADDRESS", "foo")
+    assert_trial_client()
+    monkeypatch.delenv("MYTHX_ETH_ADDRESS")
+
+
+def test_mythx_client_default_no_username_env(monkeypatch):
+    monkeypatch.setenv("MYTHX_PASSWORD", "bar")
+    assert_trial_client()
+    monkeypatch.delenv("MYTHX_PASSWORD")
+
+
+# simulate build.items() method
+BUILD_CONTAINER = {
+    "foo": {"sourcePath": "foo path", "contractName": "foo contract", "type": "contract"},
+    "bar": {"sourcePath": "bar path", "contractName": "bar contract", "type": "contract"},
+    "baz": {"sourcePath": "baz path", "contractName": "baz contract", "type": "library"},
+}
+
+
+def test_contract_locations_from_build():
+    assert get_contract_locations(BUILD_CONTAINER) == {
+        "foo path": "foo contract",
+        "bar path": "bar contract",
+        "baz path": "baz contract",
+    }
+
+
+def test_contract_types_from_build():
+    assert get_contract_types(BUILD_CONTAINER) == ({"foo", "bar"}, {"baz"})
+
+
+def test_contract_job_assembly():
+    contracts = {"SafeMath"}
+    build = mock.MagicMock()
+    build.get = lambda x: TEST_ARTIFACT
+
+    job_data = assemble_contract_jobs(build, contracts)
+    assert len(job_data) == 1
+    assert job_data["SafeMath"] == construct_request_from_artifact(TEST_ARTIFACT)
+
+
+def test_contract_job_updates():
+    contracts = {"SafeMath"}
+    libraries = {"SafeMath2"}
+
+    build = mock.MagicMock()
+    build.get = lambda x: TEST_ARTIFACT
+
+    job_data = assemble_contract_jobs(build, contracts)
+
+    artifact = deepcopy(TEST_ARTIFACT)
+    artifact["contractName"] = "SafeMath2"
+    artifact["sourcePath"] = "contracts/SafeMath2.sol"
+    build.get = lambda x: artifact
+    build.get_dependents = lambda x: ["SafeMath"]
+    job_data = update_contract_jobs_with_dependencies(build, contracts, libraries, job_data)
+
+    assert len(job_data["SafeMath"]["sources"]) == 2
+    assert "contracts/SafeMath.sol" in job_data["SafeMath"]["sources"]
+    assert "contracts/SafeMath2.sol" in job_data["SafeMath"]["sources"]
+
+    assert (
+        job_data["SafeMath"]["sources"]["contracts/SafeMath.sol"]
+        == construct_source_dict_from_artifact(TEST_ARTIFACT)["contracts/SafeMath.sol"]
+    )
+    assert (
+        job_data["SafeMath"]["sources"]["contracts/SafeMath2.sol"]
+        == construct_source_dict_from_artifact(artifact)["contracts/SafeMath2.sol"]
+    )
+
+
+JOB_DATA = {
+    "SafeMath": {
+        "contract_name": "SafeMath",
+        "bytecode": "60556023600b82828239805160001a607314601657fe5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea265627a7a72315820893edda58cfc8ab942d57dfc258288727bcaf844e6397a37aefea97a87eee32464736f6c634300050b0032",
```

Comment:
could you add `# noqa E501` on this and the next few lines to it passes flake8? Then I think we can merge this!